### ISSUE TITLE
FR: Make xcframeworks static 

### ISSUE THREAD

**Issue by jesusmateos1234** [realm/realm-cocoa#7236]
Why relamSwift.xcframework ahas a mach-o type dynamic? It doesn't contain resources like images.
Could you change to static library mach-o type?

Thanks for all

@tonyarnold @timanglade @stennie @koyachi

**Comment by jesusmateos1234** [comment]
@tgoyne @jsflax

**Comment by tonyarnold** [comment]
First up, could you **please stop mentioning me in all of your issues**? I don't work for Realm, and this isn't a project that I have used for a number of years. 

Secondly, **XCFrameworks (and frameworks) are supposed to be dynamic**. If you need a statically linked variant, then perhaps Realm can provide a static library with headers? 

For reference (emphasis mine): 

> The executable code in a framework bundle is a **dynamically linked, shared library—or, simply, a dynamic shared library**. This is a library whose code can be shared by multiple concurrently running programs.

See: https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPFrameworks/Concepts/FrameworkBinding.html

**Comment by jesusmateos1234** [comment]
Hi @tonyarnold, first of all sorry about mentioned you again, that's not will be happen other time.

Secondly, Frameworks could be static or dynamic, the mains difference between them its that dynamic frameworks live out of the exe, but, statics lives inside of your exe and no has own bundle. Using one or other could makes differences in launching time or even in app size.

Realm framework no has resources, so maybe could be possible to go without the framework bundle.

As you emphasis "**framework bundle** is a dynamically linked"

**Comment by tonyarnold** [comment]
You're correct: linking statically can make a difference to startup time on resource constrained platforms like iOS, tvOS and watchOS. 

> As you emphasis "**framework bundle** is a dynamically linked"

A framework is, by definition, a bundle containing a dylib. If you pull the binary library out of the framework bundle, it's just a library.

Framework bundles containing a static library can work, but they're unsupported by Apple/Xcode. 

Best of luck with your request - I hope you receive a favourable outcome.

**Comment by jesusmateos1234** [comment]
Static and dynamic frameworks are supported by Apple/XCode, other organizations like Facebook has xcframeworks with static frameworks inside https://github.com/facebook/facebook-ios-sdk/releases/tag/v9.3.0. The only thing that changes it's the mach-o type in build settings. If you open the XCFramework you could see frameworks no libraries. 

By in addition, i'll recommend you watch videos from Essential Developers that explain that on an extended way.

Have a good day :)

**Comment by jesusmateos1234** [comment]
@dianaafanador3 of course!! 
I think that we need to investigate if all zips are needed.

For example 

![image](https://user-images.githubusercontent.com/68072718/117838307-84480a00-b27a-11eb-98b1-317b390fe6bb.png)

Maybe that assets are not needed, because someone that need integrate manually not need this zips. I think, that someone needs an example of how to use Realm, They could download the example from the repo. Only a static XCFramework for the last xcode version that you support is needed.

How do you think about?

**Comment by jesusmateos1234** [comment]
I'm agree @dianaafanador3

**Comment by jesusmateos1234** [comment]
@dianaafanador3
